src/Entity/OgMembership.php

```diff
@@ -320,6 +320,22 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException('OG membership can not be created for an empty or anonymous user.');
     }
 
+    if (!$this->get('entity_id')->value) {
+      // Group was not set.
+      throw new \LogicException('Membership cannot be set for an empty or an unsaved group.');
```

Comment:
> If code doesn't call validate(), it is choosing to skip validation

Validation here is also a matter of security. If membership is created for anon users, it means that anons could have access to group's data/ permissions.

Or creating an `OgMembership` for a non existing group.

So, I'm ok with moving to validation, but I think we need to make sure to invoke it by default from `preSave()` to prevent the above cases. Sounds right?